testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestAvroSchemaEvolution.java

```diff
@@ -0,0 +1,285 @@
+                .containsExactlyInOrder(
+                        prepareShowColumnsResultRows(
+                                row("string_col", "varchar", "", ""),
+                                row("int_col", "integer", "", "")));
+        assertThat(query(selectStarStatement))
+                .containsExactlyInOrder(
+                        createRow(0, "stringA0", 0));
+
+        onHive().executeQuery(format("ALTER TABLE %s UNSET TBLPROPERTIES('avro.schema.url')", tableName));
+        assertThat(query(columnsInTableStatement))
+                .containsExactlyInOrder(
+                        prepareShowColumnsResultRows(
+                                row("dummy_col", "varchar", "", "")));
+    }
+
+    protected void alterTableSchemaTo(String schema)
+    {
+        onHive().executeQuery(format("ALTER TABLE %s SET TBLPROPERTIES('avro.schema.url'='%s')", tableName, schema));
+    }
+
+    private void insertData(String tableName, int rowNo, String... sqlValues)
+    {
+        String columnValues = Joiner.on(", ").join(sqlValues) +
+                (partitions.isEmpty() ? "" : ", " + getPartitionsAsListString(p -> "'" + p.getValueForRow(rowNo) + "'"));
+        query(format("INSERT INTO %s VALUES (%s)", tableName, columnValues));
+    }
+
+    private Row createRow(int rowNumber, Object... data)
+    {
+        List<Object> rowData = new ArrayList<>(Arrays.asList(data));
+        partitions.forEach(partition -> rowData.add(partition.getValueForRow(rowNumber)));
+        return new Row(rowData);
+    }
+
+    private List<Row> prepareShowColumnsResultRows(Row... rows)
+    {
+        ImmutableList.Builder<Row> rowsWithPartitionsBuilder = new ImmutableList.Builder();
+        rowsWithPartitionsBuilder.add(rows);
+        partitions.stream()
+                .map(partition -> row(partition.getName(), partition.getType(), "partition key", ""))
+                .forEach(rowsWithPartitionsBuilder::add);
+        return rowsWithPartitionsBuilder.build();
+    }
+
+    private String getPartitionsAsListString(Function<Partition, String> partitionMapper)
+    {
+        return partitions.stream()
+                .map(partitionMapper)
+                .collect(Collectors.joining(", "));
+    }
+
+    protected static class Partition
+    {
+        private String name;
+        private String type;
+
+        public Partition(String name, String type)
+        {
+            this.name = requireNonNull(name, "name is null");
+            this.type = requireNonNull(type, "type is null")
+                    .toLowerCase(ENGLISH);
+        }
+
+        public String getName()
+        {
+            return name;
+        }
+
+        public String getType()
+        {
+            return type;
+        }
+
+        public String getColumnDeclaration()
+        {
+            return name + " " + type;
+        }
+
+        public String getValueForRow(int rowNo)
+        {
+            return name + "_" + rowNo;
```

Comment:
It would only work with varchar. Given that why do we need parametrization of class on `type`?

Reply:
Will use directly string and renam constructor parameter to varcharPartitionColumns